web/satellite/src/components/project/paymentMethods/CardComponent.vue

```diff
@@ -2,108 +2,121 @@
         </div>
+    </div>
 </template>
 
 <script lang="ts">
-    import { Component, Vue } from 'vue-property-decorator';
+    import { Component, Prop, Vue } from 'vue-property-decorator';
     import Button from '@/components/common/Button.vue';
     import { APP_STATE_ACTIONS, } from '@/utils/constants/actionNames';
     import DeletePaymentMethodDialog from '@/components/project/paymentMethods/DeletePaymentMethodDialog.vue';
     import MakeDefaultPaymentMethodDialog from '@/components/project/paymentMethods/MakeDefaultPaymentMethodDialog.vue';
 
     @Component({
-        props: {
-            paymentMethod: {
-                type: Object,
-                default: {}
-            },
-        },
-        methods: {
-            formatDate: function (d: string): string {
-                return new Date(d).toLocaleDateString('en-US', {timeZone: 'UTC'});
-            },
-            onMakeDefaultClick: async function () {
-                if ((this as any).getSetDefaultPaymentMethodID == this.$props.paymentMethod.id) {
-                    this.$store.dispatch(APP_STATE_ACTIONS.CLOSE_POPUPS);
-
-                    return;
-                }
-
-                this.$store.dispatch(APP_STATE_ACTIONS.SHOW_SET_DEFAULT_PAYMENT_METHOD_POPUP, this.$props.paymentMethod.id);
-            },
-            onDeletePaymentMethodClick: async function() {
-                if ((this as any).getDeletePaymentMethodID == this.$props.paymentMethod.id) {
-                    this.$store.dispatch(APP_STATE_ACTIONS.CLOSE_POPUPS);
-
-                    return;
-                }
-
-                this.$store.dispatch(APP_STATE_ACTIONS.SHOW_DELETE_PAYMENT_METHOD_POPUP, this.$props.paymentMethod.id);
-            }
-        },
-        computed: {
-            getDeletePaymentMethodID: function(): string {
-                return this.$store.state.appStateModule.appState.deletePaymentMethodID;
-            },
-            getSetDefaultPaymentMethodID: function(): string {
-                return this.$store.state.appStateModule.appState.setDefaultPaymentMethodID;
-            },
-            isDeletePaymentMethodPopupShown: function (): boolean {
-                return this.$store.state.appStateModule.appState.deletePaymentMethodID == this.$props.paymentMethod.id;
-            },
-            isSetDefaultPaymentMethodPopupShown: function(): boolean {
-                return this.$store.state.appStateModule.appState.setDefaultPaymentMethodID == this.$props.paymentMethod.id;
-            },
-        },
         components: {
             MakeDefaultPaymentMethodDialog,
             Button,
             DeletePaymentMethodDialog,
         }
     })
-    export default class CardComponent extends Vue {}
+    export default class CardComponent extends Vue {
+        @Prop({default: {}})
+        private readonly paymentMethod: PaymentMethod;
+
+        public formatDate(d: string): string {
```

Comment:
lets add this function ad PaymentMethod level.

In this case our view becomes more lightweight and usage of it will be easier

F.e. paymentMethod.addedAtFormatted() instead of formatDate(paymentMethod.addedAt)

Reply:
this will be solved in @Barterio's PR